test/gen_casadi_test.py

```diff
@@ -2420,6 +2426,36 @@ def test_iterative_simplification_large(self):
         self.assertEqual(12, len(casadi_model.alg_states))
 
         casadi_model.simplify(compiler_options)
+        
+    def test_factor_and_simplify_binary_ops(self):
+        # The initial equation is encountered first, and should properly
+        # initialize the corresponding symbol for the derivative.
+        txt = """
+                model FactorAndSimplify
+                    Real y;
+                    Real x;
+                    Real z;
+                equation
+                    (-x) - y = 0;
+                    3*y = 0;
+                    z/9 = 0;
+                end FactorAndSimplify;
+            """
+
+        ast_tree = parser.parse(txt)
+
+        casadi_model = gen_casadi.generate(ast_tree, 'FactorAndSimplify')
+        casadi_model.simplify({'factor_and_simplify_equations': True})
+
+        ref_model = Model()
+
+        y = ca.MX.sym('y')
+        x = ca.MX.sym('x')
+        z = ca.MX.sym('z')
+
+        ref_model.alg_states = [Variable(y), Variable(x), Variable(z)]
+        ref_model.equations = [x + y, z, y]
 
+        self.assert_model_equations_equivalent(ref_model, casadi_model)
```

Comment:
Why is `assert_model_equivalent` not good enough? Also, please keep two blank lines before the `if __name___` part :)

Reply:
Sorry about the black lines eheh :) with pycharm I have no excuses :)) I'll fix it!

As for `assert_model_equivalent` I think it is not enough because only the variables are compared (we could maybe change the name because it could be misleading).

I could definitely use `assert_model_equivalent_numeric` but in this case also the order of the equations is compared.

What do you think?